daemon/src/maker_inc_connections.rs

```diff
@@ -441,6 +377,119 @@ impl Actor {
     }
 }
 
+/// Sets up a new connection on the given stream including all relevant background tasks like
+/// heartbeat and reading messages from the stream.
+async fn setup_new_connection(
+    this: Address<Actor>,
+    noise_priv_key: x25519_dalek::StaticSecret,
+    heartbeat_interval: Duration,
+    stream: TcpStream,
+    address: SocketAddr,
+) {
+    let (mut read, write, taker_id) = match upgrade(stream, noise_priv_key).await {
+        Ok((read, write, identity)) => (read, write, identity),
+        Err(e) => {
+            tracing::warn!(address = %address, "Failed to upgrade incoming connection: {:#}", e);
+            return;
+        }
+    };
+
+    let mut tasks = Tasks::default();
+    tasks.add({
+        let this = this.clone();
+
+        async move {
+            while let Ok(Some(msg)) = read.try_next().await {
+                let res = this.send(FromTaker { taker_id, msg }).await;
+
+                if res.is_err() {
+                    break;
+                }
+            }
+
+            let _ = this.send(ReadFail(taker_id)).await;
+        }
+    });
+    tasks.add({
+        let this = this.clone();
+
+        async move {
+            while let Ok(()) = this.send(SendHeartbeat(taker_id)).await {
+                tokio::time::sleep(heartbeat_interval).await;
+            }
+        }
```

Comment:
How is connection cleanup handled? 
Given that we can e.g. not send the heartbeat we just end this task here. How would that effect the inserted connection?